File: medico/Services/Medico.Api/ClientApp/src/app/_models/template.ts
export class Template {
    id: string;
    companyId: string;
    templateOrder: number;
    title: string;
    reportTitle: string;
    detailedTemplateHtml: string;
    initialDetailedTemplateHtml: string;
    defaultTemplateHtml: string;
    isRequired: boolean;
    isActive: boolean;
    isHistorical: boolean;
    templateTypeId: string;
    libraryTemplateId: string | null;
    version: number | null;

    constructor() {
        this.isActive = true;
        this.isRequired = false;
        this.isHistorical = false;
    }
}

File: medico/Domain/Medico.Domain/Models/Template.cs
using System;
using System.Collections.Generic;

namespace Medico.Domain.Models
{
    public class Template : Entity
    {
        public Guid? LibraryTemplateId { get; set; }

        public Template LibraryTemplate { get; set; }

        public int? Version { get; set; }

        public bool IsActive { get; set; }

        public Guid? CompanyId { get; set; }

        public Company Company { get; set; }

        public int? TemplateOrder { get; set; }

        public string Title { get; set; }

        public string ReportTitle { get; set; }

        public string DetailedTemplateHtml { get; set; }

        public string InitialDetailedTemplateHtml { get; set; }

        public string DefaultTemplateHtml { get; set; }

        public bool IsRequired { get; set; }

        public bool IsHistorical { get; set; }

        public Guid TemplateTypeId { get; set; }

        public TemplateType TemplateType { get; set; }

        public List<ChiefComplaintTemplate> ChiefComplaintTemplates { get; set; }

        public List<Template> LibraryRelatedTemplates { get; set; }

        public List<TemplateSelectableList> TemplateSelectableLists { get; set; }
        
        public List<TemplateExpression> TemplateExpressions { get; set; }
    }
}